src/msg/consumer/consumer.go

```diff
@@ -257,7 +261,10 @@ func (c *consumer) Close() {
 	}
 	c.closed = true
 	c.Unlock()
-
+	// Release the bufio.NewWriterSize() and bufio.NewReaderSzie()
+	c.m.close.Inc(1)
+	c.w.Reset(nil)
+	c.decoder.ResetReader(nil)
```

Comment:
`ResetReader()` is only called in test files. I suspect this is a real bug and it may be already fixed in Chronosphere's internal codebase.

https://github.com/m3db/m3/search?q=ResetReader